src/query_plan/query_planner.rs

```diff
@@ -106,26 +120,305 @@ impl Default for QueryPlannerDebugConfig {
     fn default() -> Self {
         Self {
             bypass_planner_for_single_subgraph: false,
-            max_evaluated_plans: 10000,
+            max_evaluated_plans: 10_000,
             paths_limit: None,
         }
     }
 }
 
 pub struct QueryPlanner {
-    config: Arc<QueryPlannerConfig>,
+    // TODO(@goto-bus-stop) not convinced that these arcs are all necessary
+    config: QueryPlannerConfig,
     federated_query_graph: Arc<QueryGraph>,
     supergraph_schema: ValidFederationSchema,
     api_schema: ValidFederationSchema,
     subgraph_federation_spec_definitions: Arc<IndexMap<NodeStr, &'static FederationSpecDefinition>>,
```

Comment:
I didn't find this in the JS implementation. I do understand what it needs to do because of the type. But how is it going to be used?